rpc/json/ws.go

```diff
@@ -33,13 +33,40 @@ func (wsc *wsConn) sendLoop() {
 	}
 }
 
+func CheckOrigin(r *http.Request, allowedOrigins []string) bool {
+	origin := r.Header.Get("Origin")
+	if origin == "" {
+		return true
+	}
+
+	if len(allowedOrigins) > 0 && allowedOrigins[0] == "*" {
+		return true
+	}
+	for _, allowed := range allowedOrigins {
+		if allowed == origin {
+			return true
+		}
+	}
+	return false
+}
+
+func OriginChecker(allowedOrigins []string) func(*http.Request) bool {
+	return func(r *http.Request) bool {
+		return CheckOrigin(r, allowedOrigins)
+	}
+}
+
 func (h *handler) wsHandler(w http.ResponseWriter, r *http.Request) {
 	// TODO(tzdybal): configuration options
 	upgrader := websocket.Upgrader{
 		ReadBufferSize:  1024,
 		WriteBufferSize: 1024,
 	}
 
+	// TODO: use CORSAllowedMethods, CORSAllowedHeaders
```

Comment:
what does this TODO mean? in future this will be changed?

Reply:
The config has these fields, we can either strictly follow the field names and only allow certain methods or headers.

But I looked at tendermint and they are basically just skipping these config fields, they're not used at all, see: https://github.com/cometbft/cometbft/blob/b0d64fa2f27d2439e655338117eb4eaa250a156c/rpc/jsonrpc/server/ws_handler.go#L49-L59

I'd say that tendermint config is actually misleading as people might think their RPC is gated by certain methods/headers while it's not. It's not a big security issue but still something that can catch off guard.